modules/index.js

```
@@ -14,26 +14,73 @@
-// eslint-disable-next-line no-use-before-define, max-len
-const mediaMO = window.MutationObserver ? new MutationObserver(mediaMCB)
-  /* c8 ignore next */ : {};
+// eslint-disable-next-line no-use-before-define
+const mediaMO = createMO(mediaMCB);
+
+// Check for the presence of URL parameters
+const hasUrlParameters = ({ urlParameters }) => urlParameters.keys().length > 0;
+// Check for the presence of a given cookie
+const hasCookieKey = (key) => () => document.cookie.split(';').map((c) => c.trim()).some((cookie) => cookie.startsWith(`${key}=`));
+
+const pluginBasePath = new URL('.rum/@adobe/helix-rum-enhancer@^2/src/plugins', sampleRUM.baseURL).href;
+
+const PLUGINS = {
+  cwv: `${pluginBasePath}/cwv.js`,
+  navigation: `${pluginBasePath}/navigation.js`,
+  // Interactive elements
+  form: { url: `${pluginBasePath}/form.js`, condition: () => document.body.querySelector('form') },
+  video: { url: `${pluginBasePath}/video.js`, condition: () => document.body.querySelector('video') },
+  // Martech
+  ads: { url: `${pluginBasePath}/ads.js`, condition: hasUrlParameters },
+  email: { url: `${pluginBasePath}/email.js`, condition: hasUrlParameters },
+  onetrust: { url: `${pluginBasePath}/onetrust.js`, condition: () => hasCookieKey('OptanonAlertBoxClosed') },
+  utm: { url: `${pluginBasePath}/utm.js`, condition: hasUrlParameters },
+};
+
+const allPlugins = {
+  ...(window.RUM_PLUGINS || {}),
+  PLUGINS,
+};
+
+const PLUGIN_PARAMETERS = {
+  context: document.body,
+  fflags,
+  sampleRUM,
+  sourceSelector,
+  targetSelector,
+};
+
+const pluginCache = new Map();
+
+async function loadPlugin(key, params) {
+  const plugin = allPlugins[key];
+  if (!plugin) return Promise.reject(new Error(`Plugin ${key} not found`));
+  const usp = new URLSearchParams(window.location.search);
+  if (!pluginCache.has(key) && plugin.condition && !plugin.condition({ urlParameters: usp })) {
+    return Promise.reject(new Error(`Condition for plugin ${key} not met`));
```

Why Promise reject instead of a simple `throw`? Given that nobody will handle the error, we could also simply log it or ignore it.

---

Yes, this is still pretty raw… I was thinking going with the approach that takes the fewest bytes in the end

---

Silently ignoring errors wins in that department.

---

Added a catch clause in the caller